icon-validator/validate-icon.c

```diff
@@ -85,6 +91,17 @@ validate_icon (const char *arg_width,
       return 1;
     }
 
+  if (width != height)
+    {
+      g_printerr ("Expected a square icon but got: %dx%d\n", width, height);
+      return 1;
+    }
+
+  /* Print the format and size for consumption by (at least) the dynamic
+   * launcher portal. xdg-desktop-portal has a copy of this file.
+   */
+  g_print ("%s:%d", name, width);
```

Comment:
I think I'd prefer this to come out in a format that could be extended to print other fields without that being an incompatible change, perhaps one of:

* `key=value` lines
* `key=value\0` binary
* JSON
* GKeyFile (ugh... but GLib has a semi-high-quality parser for it built-in, so...)
* the GMarkup subset of XML (twice as much ugh)
* GVariant

The sort of thing I'm thinking of is: if at some point we added an `--allow-non-square` option, then we'd have to invent a way to print the width and height separately. Or, if we wanted to print the colour depth, or the MIME type, or some other random fact about the icon that can't be learned without parsing it, it would be good if we had a way to add that in a way that won't require callers to change how they parse the output.

For example, if the output now was

```
format=png
width=64
```

then that could turn into

```
format=png
width=64
height=60
mime=image/png
bpp=15
contrast-ratio=0.42
```

if it ever became necessary.

Reply:
`GKeyFile` seems like a good option, since it's simple to use, simple to debug, and it's reasonable to expect that anything using this has or can gain a dependency on GLib.

Reply:
Pushed the changes needed to use `GKeyFile` and I'll do the same in x-d-p.